app/views/rooms/_form.html.erb

```diff
@@ -151,3 +168,29 @@
     </div>
   </div>
 <% end %>
+
+
+<script>
+  document.getElementById('copy-icon-group').addEventListener('click', async function() {
+    let useSharedCodeCheckbox = document.getElementById('use_shared_code_checkbox');
+    
+    let inputField = document.querySelector('#shared_code_field');
+    inputField.removeAttribute('disabled');
+    let copyText = document.querySelector('#shared_code_field').select();
+    document.execCommand('copy');
+    window.getSelection().removeAllRanges();
+    if (!useSharedCodeCheckbox.checked) {
+      inputField.setAttribute('disabled', 'true');
+    }
+    let copyIcon = document.querySelector('.copy-icon');
+    copyIcon.src = '<%= asset_path("check.svg") %>';
+
+    let copiedText = document.getElementById('copied-text');
+    copiedText.style.display = 'inline';
+
+    setTimeout(function() {
+      copyIcon.src = '<%= asset_path("copy.svg") %>';
+      copiedText.style.display = 'none';
+    }, 2000);
+  });
+</script>
```

Comment:
we keep all js code in js files. you can create a new one under javascript/packs or add it to edit.js